docs/StardustDocs/topics/operations/multiple/join.md

```diff
@@ -0,0 +1,393 @@
+
+<inline-frame src="./resources/notebook_test_join_15.html" width="100%" height="500px"></inline-frame>
+
+#### Filter {collapsible="true"}
+
+<!---FUN notebook_test_join_13-->
+
+```kotlin
+dfLeft
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_13.html" width="100%" height="500px"></inline-frame>
+
+<!---FUN notebook_test_join_14-->
+
+```kotlin
+dfRight
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_14.html" width="100%" height="500px"></inline-frame>
+
+
+<!---FUN notebook_test_join_16-->
+
+```kotlin
+// FILTER JOIN:
+// Keep ONLY left rows that have ANY match on (name, city).
+// No right-side columns are added.
+dfLeft.filterJoin(dfRight) { name and city }
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_16.html" width="100%" height="500px"></inline-frame>
+
+#### Left {collapsible="true"}
+
+<!---FUN notebook_test_join_13-->
+
+```kotlin
+dfLeft
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_13.html" width="100%" height="500px"></inline-frame>
+
+<!---FUN notebook_test_join_14-->
+
+```kotlin
+dfRight
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_14.html" width="100%" height="500px"></inline-frame>
+
+
+<!---FUN notebook_test_join_17-->
+
+```kotlin
+// LEFT JOIN:
+// Keep ALL left rows. If (name, city) matches, attach right columns;
+// if not, right columns are null (e.g., Alice–London has no right match).
+dfLeft.leftJoin(dfRight) { name and city }
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_17.html" width="100%" height="500px"></inline-frame>
+
+#### Right {collapsible="true"}
+
+<!---FUN notebook_test_join_13-->
+
+```kotlin
+dfLeft
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_13.html" width="100%" height="500px"></inline-frame>
+
+<!---FUN notebook_test_join_14-->
+
+```kotlin
+dfRight
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_14.html" width="100%" height="500px"></inline-frame>
+
+
+<!---FUN notebook_test_join_18-->
+
+```kotlin
+// RIGHT JOIN:
+// Keep ALL right rows. If no left match, left columns become null
+// (e.g., Alice with city=null exists only on the right).
+dfLeft.rightJoin(dfRight) { name and city }
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_18.html" width="100%" height="500px"></inline-frame>
+
+#### Full {collapsible="true"}
+
+<!---FUN notebook_test_join_13-->
+
+```kotlin
+dfLeft
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_13.html" width="100%" height="500px"></inline-frame>
+
+<!---FUN notebook_test_join_14-->
+
+```kotlin
+dfRight
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_14.html" width="100%" height="500px"></inline-frame>
+
+
+<!---FUN notebook_test_join_19-->
+
+```kotlin
+// FULL JOIN:
+// Keep ALL rows from both sides. Where there's no match on (name, city),
+// the other side is filled with nulls.
```

Comment:
could you make all `nulls` bold? so it's clearer these are new, added to fill in the gaps

Reply:
`.format().with { if (it == null) bold else null }` will do

I still find the examples hard to follow, I'm wondering how we can make it as clear as possible :)